### PR TITLE
etcd: Build image 3.5.11

### PR DESCRIPTION
**Pull request by mzaian**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

Build Image for [v3.5.11](https://github.com/etcd-io/etcd/releases/tag/v3.5.11).

**Which issue(s) this PR fixes**:
Fixes NONE #

**Does this PR introduce a user-facing change?**:
```
NONE
```

```release-note
etcd: build image for v3.5.11
```

### CODE REVIEW

**Review by xmudrii — APPROVED** [review]
/lgtm
/approve

File: cluster/images/etcd/Makefile [hunk 1]
```diff
@@ -83,7 +83,7 @@ endif
 # This option is for running docker manifest command
 export DOCKER_CLI_EXPERIMENTAL := enabled
 # golang version should match the golang version of the official build from https://github.com/etcd-io/etcd/releases.
-GOLANG_VERSION := 1.20.10
+GOLANG_VERSION := 1.20.11
```
Comment (serathius) [inline cluster/images/etcd/Makefile:86]: Just before the release we updated golang to 1.20.12 https://github.com/etcd-io/etcd/blob/v3.5.11/.go-version
```suggestion
GOLANG_VERSION := 1.20.12
```

Comment (mzaian) [reply cluster/images/etcd/Makefile:86]: Done

**Review by SataQiu — COMMENTED** [review]
/lgtm

**Pull request merged**